Std/Data/BitVec/Basic.lean

```diff
@@ -316,9 +310,18 @@ Bitwise NOT for bit vectors.
 ```
 SMT-Lib name: `bvnot`.
 -/
-protected def not (x : BitVec n) : BitVec n := -(x + .ofNat n 1)
+protected def not (x : BitVec n) : BitVec n :=
+  .ofFin ⟨(1 <<< n).pred,
+    n.one_shiftLeft.symm ▸
+      (Nat.pred_lt <| Nat.ne_of_gt <| Nat.pos_pow_of_pos _ <| Nat.zero_lt_succ _)⟩ ^^^ x
```

Comment:
Let's go forward with this suggestion, to have the `ones` term as a let. This PR is blocking https://github.com/leanprover-community/mathlib4/pull/8711 so I would prefer we merge it sooner rather than later.
If it turns out later that we find another use-case for this `ones` term, we can always extract it out to a separate def then.